packages/firebase_database/ios/Classes/FirebaseDatabasePlugin.m

```diff
@@ -70,7 +70,12 @@ - (NSDictionary *)dictionary {
   }
   id equalTo = parameters[@"equalTo"];
   if (equalTo) {
-    query = [query queryEqualToValue:equalTo];
+    id equalToKey = parameters[@"equalToKey"];
+    if (equalToKey) {
```

Comment:
Should we not check for equality with `[NSNull null]` here?

Reply:
I simply updated this block to follow the same pattern as in `startAt` and `endAt` blocks where this check seems to be working as expected. I'm not an Obj-C expert by any means though. 

This [SO answer](https://stackoverflow.com/a/28852775) says that it's a right way to check for `nil` (which is a [literal null value for Obj-C objects](http://nshipster.com/nil/)).

Reply:
You are right, because you also took care to change the Dart code to ensure that if there is an `equalToKey` entry in `parameters`, then it maps to a non-null value. I just didn't see that on my first reading.